src/cli.rs

```diff
@@ -1,27 +1,415 @@
-use std::env;
 use std::path::{Path, PathBuf};
+use std::{env, fmt};
 
 use crate::cargo::Subcommand;
 use crate::errors::Result;
 use crate::file::{absolute_path, PathExt};
 use crate::rustc::TargetList;
-use crate::shell::{self, MessageInfo};
+use crate::shell::{MessageInfo, COLORS};
 use crate::Target;
 
+/// An option for a CLI flag value.
+#[derive(Debug)]
+pub enum FlagOption<T> {
+    /// Value for flag was not provided.
+    Some(T),
+    /// Flag was not provided.
+    None,
+    /// Flag was provided multiple times.
+    Double,
+    /// Flag was not provided.
+    Missing,
+}
+
+impl<T> FlagOption<T> {
+    pub fn is_some(&self) -> bool {
+        matches!(self, Self::Some(_))
+    }
+
+    pub fn is_none(&self) -> bool {
+        matches!(self, Self::Some(_))
+    }
+
+    /// If the flag is present multiple times, always set to `Double`.
```

Comment:
this doc comment describes the behaviour for a certain case, but not what it does